src/test/cluster.zig

```diff
@@ -174,27 +247,59 @@ pub const Cluster = struct {
         return cluster;
     }
 
-    pub fn destroy(cluster: *Cluster) void {
+    pub fn deinit(cluster: *Cluster) void {
+        cluster.storage_checker.deinit();
+        cluster.state_checker.deinit();
         cluster.network.deinit();
-
+        for (cluster.clients) |*client| client.deinit(cluster.allocator);
+        for (cluster.client_pools) |*pool| pool.deinit(cluster.allocator);
         for (cluster.replicas) |*replica| replica.deinit(cluster.allocator);
-        cluster.allocator.free(cluster.replicas);
-        cluster.allocator.free(cluster.health);
-        for (cluster.pools) |*pool| pool.deinit(cluster.allocator);
-        cluster.allocator.free(cluster.pools);
-
+        for (cluster.replica_pools) |*pool| pool.deinit(cluster.allocator);
         for (cluster.storages) |*storage| storage.deinit(cluster.allocator);
-        cluster.allocator.free(cluster.storages);
 
+        cluster.allocator.free(cluster.clients);
+        cluster.allocator.free(cluster.client_pools);
+        cluster.allocator.free(cluster.replicas);
+        cluster.allocator.free(cluster.replica_health);
+        cluster.allocator.free(cluster.replica_pools);
+        cluster.allocator.free(cluster.storages);
+        cluster.allocator.destroy(cluster.storage_fault_atlas);
+        cluster.allocator.destroy(cluster.network);
         cluster.allocator.destroy(cluster);
     }
 
+    pub fn tick(cluster: *Cluster) void {
+        cluster.network.tick();
+
+        for (cluster.clients) |*client| client.tick();
+        for (cluster.storages) |*storage| storage.tick();
+        for (cluster.replicas) |*replica, i| {
+            switch (cluster.replica_health[i]) {
+                .up => replica.tick(),
+                // Keep ticking the time so that it won't have diverged too far to synchronize
```

Comment:
If the clock loses synchronization in the simulation, the cluster can't recover and the test will never finish (it will hit the `ticks_max` limit). In a real-world cluster NTP guards against (or recovers from) clock desync.